tests/client/amo/containers/TestApp.js

```diff
@@ -141,6 +143,19 @@ describe('App', () => {
     assert.ok(dispatch.calledWith({ type: INSTALL_STATE, payload }));
   });
 
+  it('sets up a callback for setting add-on status', () => {
+    const dispatch = sinon.spy();
+    const { setUserAgent } = mapDispatchToProps(dispatch);
+    const userAgent = 'tofubrowser';
+    const { browser, os } = UAParser(userAgent);
+
+    setUserAgent(userAgent);
+    assert.ok(dispatch.calledWith({
```

Comment:
I think it would be better to stub out `setUserAgentAction` and just assert that `dispatch` gets called with the result of your stub `setUserAgentAction`. This will allow the `setUserAgentAction` function to evolve independently of the `App` component.